src/plugins/explore/public/components/visualizations/bar_gauge/bar_gauge_utils.ts

```diff
@@ -0,0 +1,179 @@
+    yAxis,
+    yAxisStyle: isXNumerical ? axisStyle : nullStyle,
+  };
+};
+
+export const thresholdsToGradient = (thresholds: Threshold[]) => {
+  return thresholds.map((threshold: Threshold, index) => {
+    return {
+      calculate: `${threshold.value}`,
+      as: `threshold${index}`,
+    };
+  });
+};
+
+export const symbolOpposite = (orientationMode: string, symbol: string) => {
+  if (orientationMode === 'horizontal') {
+    return symbol === 'x' ? 'y' : 'x';
+  }
+  return symbol;
+};
+
+export const getGradientConfig = (
+  orientationMode: string,
+  displayMode: string,
+  isXaxisNumerical: boolean
+) => {
+  if (
+    (!isXaxisNumerical && orientationMode === 'horizontal') ||
+    (isXaxisNumerical && orientationMode !== 'horizontal')
+  ) {
+    if (displayMode === 'gradient')
+      return {
+        x1: 0,
+        y1: 0,
+        x2: 1,
+        y2: 0,
+      };
+  }
+
+  if (displayMode === 'gradient')
+    return {
+      x1: 1,
+      y1: 1,
+      x2: 1,
+      y2: 0,
+    };
+};
+
+export const darkenColor = (hex: string, degree = 1) => {
+  // degree: 1 = 10%, 2 = 20%, etc.
+  const factor = 1 - degree * 0.1;
+
+  if (hex.length === 4) {
+    hex = '#' + hex[1] + hex[1] + hex[2] + hex[2] + hex[3] + hex[3];
+  }
+
+  let r = parseInt(hex.slice(1, 3), 16);
+  let g = parseInt(hex.slice(3, 5), 16);
+  let b = parseInt(hex.slice(5, 7), 16);
+
+  r = Math.max(Math.floor(r * factor), 0);
+  g = Math.max(Math.floor(g * factor), 0);
+  b = Math.max(Math.floor(b * factor), 0);
+
+  const toHex = (n: number) => n.toString(16).padStart(2, '0');
+  return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
+};
+
+export const processThresholds = (thresholds: Threshold[]) => {
+  const result: Threshold[] = [];
+
+  for (let i = 0; i < thresholds.length; i++) {
+    const current = thresholds[i];
+    const next = thresholds[i + 1];
+
+    // if the next threshold has the same value, use next
+    if (next && next.value === current.value) continue;
+
+    result.push(current);
+  }
+
+  return result;
+};
+
+export const NormalizeData = (data: number, start: number, end: number) => {
+  // normalize data value between start and end into 0–1 range
+  return (data - start) / (end - start);
+};
+
+export const generateParams = (
+  thresholds: Threshold[],
+  styleOptions: BarGaugeChartStyle,
+  isXaxisNumerical: boolean
+) => {
+  const result: any[] = [];
+
+  for (let i = 0; i < thresholds.length; i++) {
+    const start = thresholds[0].value;
+
+    const end = thresholds[i].value;
+
+    if (i === 0) {
+      result.push({
+        name: `test${i}`,
+        value: {
+          gradient: 'linear',
+          ...getGradientConfig(
+            styleOptions.exclusive.orientation,
+            styleOptions.exclusive.displayMode,
+            isXaxisNumerical
+          ),
+          stops: [
+            { offset: 0, color: thresholds[0].color },
+            {
+              offset: 1,
+              color: darkenColor(thresholds[0]?.color, 2),
```

Comment:
I think we don't need to darken the color, just use `thresholds[0].color` in this case